cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1493,47 +1493,63 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 		return err
 	}
 
-	// 2. mount ubuntu-seed
-	// use the disk we mounted ubuntu-boot from as a reference to find
-	// ubuntu-seed and mount it
-	partUUID, err := disk.FindMatchingPartitionUUIDWithFsLabel("ubuntu-seed")
+	// 1.1. measure model
+	err = stampedAction("run-model-measured", func() error {
+		return secbootMeasureSnapModelWhenPossible(mst.UnverifiedBootModel)
+	})
 	if err != nil {
 		return err
 	}
 
-	// fsck is safe to run on ubuntu-seed as per the manpage, it should not
-	// meaningfully contribute to corruption if we fsck it every time we boot,
-	// and it is important to fsck it because it is vfat and mounted writable
-	// TODO:UC20: mount it as read-only here and remount as writable when we
-	//            need it to be writable for i.e. transitioning to recover mode
+	// The model is now measured, use it to check if this is a classic install
+	model, err := mst.UnverifiedBootModel()
+	if err != nil {
+		return err
+	}
+	isClassic := model.Classic()
+	if isClassic {
+		logger.Noticef("generating mounts for classic system, run mode")
+		boot.InitramfsWritableDir = boot.InitramfsDataDir
```

Comment:
this needs to be done somehow differently, redefining globals in other packages unless they are hooks is not something we do

Reply:
Changed, I pass around an argument where needed instead.